### ISSUE TITLE
BatchNorm Evaluation Mode Backward Fails with cudnn Enabled

### ISSUE THREAD

**Issue by zhanghang1989** [apache/incubator-mxnet#9648]
When installing MXNet with cudnn enabled ``USE_CUDNN=1``. The backward of BatchNorm operator fails

## Reproducing the error
```
import mxnet as mx
import mxnet.ndarray as F
from mxnet import autograd

B,C,H,W = 4,3,2,2
x = mx.nd.random.poisson(1,shape=(B,C,H,W)).as_in_context(mx.gpu(0))
gamma = mx.nd.random.normal(shape=(C)).as_in_context(mx.gpu(0))
beta = mx.nd.random.normal(shape=(C)).as_in_context(mx.gpu(0))
mean = mx.nd.random.normal(shape=(C)).as_in_context(mx.gpu(0))
std = mx.nd.random.normal(shape=(C)).as_in_context(mx.gpu(0))
x.attach_grad()

with autograd.record(False):
    y = F.BatchNorm(x, gamma, beta, mean, std.square(), fix_gamma=False)     
    loss=y.square().sum()
loss.backward(train_mode=False)
```
got the error:
```bash
terminate called after throwing an instance of 'dmlc::Error'
  what():  [19:23:23] src/engine/./threaded_engine.h:359: [19:23:23] src/operator/nn/./cudnn/cudnn_batch_norm-inl.h:193: Check failed: ctx.is_train && !param_.use_global_stats use global statistics is not yet supported in CuDNNBatchNorm
```

**Comment by zhanghang1989** [comment]
Backward in evaluation mode should be handled differently. Current code is here:
https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cu#L645-L656

PyTorch had a related issue https://github.com/pytorch/pytorch/issues/4284

**Comment by zhanghang1989** [comment]
I think this should be labeled as `bug`

**Comment by cjolivier01** [comment]
With what frequency can you reproduce?  I haven't been able to reproduce.

**Comment by anirudh2290** [comment]
@zhanghang1989 related pr merged. can this be closed ?

**Comment by zhanghang1989** [comment]
Reopen this due to CUDNN is handled wrongly @piiswrong

**Comment by tchaton** [comment]
```
import torch
import torch.nn as nn
import torch.nn.functional as F
import math
import numpy as np

def _upsample(x):
    h, w = x.shape[2:]
    return F.upsample_bilinear(x, size=(h * 2, w * 2))


def upsample_conv(x, conv):
    return conv(_upsample(x))

class genBlock(nn.Module):
    def __init__(self, in_channels, out_channels,
                 activation=F.relu, hidden_channels=None, ksize=3, pad=1, upsample=False, n_classes=0):
        super(genBlock, self).__init__()
        self.activation = activation
        self.upsample = upsample
        self.learnable_sc = in_channels != out_channels or upsample
        hidden_channels = out_channels if hidden_channels is None else hidden_channels
        self.n_classes = n_classes
        self.c1 = nn.Conv2d(in_channels, hidden_channels, kernel_size=ksize, padding=pad)
        #nn.init.xavier_uniform_(self.c1.weight.data, math.sqrt(2))
        self.c2 = nn.Conv2d(hidden_channels, out_channels, kernel_size=ksize, padding=pad)
        #nn.init.xavier_uniform_(self.c2.weight.data, math.sqrt(2))
        self.b1 = nn.BatchNorm2d(in_channels)
        self.b2 = nn.BatchNorm2d(hidden_channels)
        if self.learnable_sc:
            self.c_sc = nn.Conv2d(in_channels, out_channels, kernel_size=ksize, padding=pad)
    def residual(self, x):
        h = x
        h = self.b1(h)
        h = self.activation(h)
        h = upsample_conv(h, self.c1) if self.upsample else self.c1(h)
        h = self.b2(h)
        h = self.activation(h)
        h = self.c2(h)
        return h

    def shortcut(self, x):
        if self.learnable_sc:
            x = upsample_conv(x, self.c_sc) if self.upsample else self.c_sc(x)
            return x
        else:
            return x

    def forward(self, input):
        return self.residual(input) + self.shortcut(input)
if __name__== "__main__":

    noise = torch.randn(1,256, 4, 4).cuda()
    g = genBlock(256, 256, activation=F.relu, upsample=True).cuda()
    #g.apply(weights_init)
    out = g(noise)
    print(out.shape)
```
